src/utilities/identities/identities.ts

```diff
@@ -267,7 +274,13 @@ export async function getIdentityCryptoFromKeypair(
 
   function sign(plaintext: string) {
     const signature = Crypto.u8aToHex(authenticationKey.sign(plaintext));
-    const didKeyUri = didDetails.getKeyIds(KeyRelationship.authentication)[0];
+
+    const authorizationKey = didDetails.getVerificationKeys(
+      KeyRelationship.authentication,
+    )[0];
+    // TODO: replace with assembleKeyId when available
+    const didKeyUri = `${didDetails.did}#${authorizationKey.id}`;
```

Comment:
You could use `assembleKeyId` if using `DidDetails`. Here the app is signing a payload, so I assume is on the DID owner side. And for that, `DidDetails` should be used.

Reply:
I wasn’t able to fix that yet, `IDidDetails` is still used accross the board

Reply:
If you switch to `DidDetails`, then the comments above about using `.encryptionKey` etc apply. For now, if you use `IDidDetails` of course that is not possible.

Reply:
Now it should be better